assets/src/edit-story/components/library/panes/text/useInsertPreset.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useRef } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import getInsertedElementSize from '../../../../utils/getInsertedElementSize';
+import { PAGE_HEIGHT } from '../../../../constants';
+import useLibrary from '../../useLibrary';
+import { useHistory } from '../../../../app/history';
+import { dataFontEm } from '../../../../units';
+
+const POSITION_MARGIN = dataFontEm(1);
+const TYPE = 'text';
+
+function useInsertPreset() {
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+  const {
+    state: { versionNumber },
+  } = useHistory();
+
+  const lastPreset = useRef(null);
+
+  useEffect(() => {
+    // Version number change is happening due to adding a preset.
+    // If we have set the last element but not the history version number yet,
+    // Set the version number that was the result of adding the preset.
+    if (lastPreset.current?.element && !lastPreset.current.versionNumber) {
+      lastPreset.current.versionNumber = versionNumber;
+    } else if (lastPreset.current?.versionNumber) {
+      // If the version number changes meanwhile and we already have it set
+      // something else changed meanwhile so clear the lastPreset, too.
+      lastPreset.current = null;
+    }
+  }, [versionNumber]);
```

Comment:
Now you don't actually use the value of `versionNumber`, so no need to store it - it's the mere presence that's relevant. This could be changed to:

```js
  useEffect(() => {
    // Version number change is happening due to adding a preset.
    if (lastPreset.current?.element) {
      if (lastPreset.current.wasJustAdded) {
        lastPreset.current.wasJustAdded = false;
      } else {
        // If the version number changes meanwhile and we already have it set
        // something else changed meanwhile so clear the lastPreset, too.
        lastPreset.current = null;
      }
    }
  }, [versionNumber]);
```

And then default `wasJustAdded: true` further below. But I'm not sure it's a huge improvement. It's kind of less transparent, I guess?

Reply:
Yes, seems a bit more self-explanatory with the version numbers being used. I'll leave it in for now.